aten/src/ATen/native/mps/operations/Convolution.mm

```diff
@@ -158,57 +222,77 @@ Tensor _mps_convolution_impl(const Tensor& input_t,
-                                                                    dimension:-3
-                                                                withDimension:-4
-                                                                         name:nil];
-            outputTensor = [mpsGraph depthwiseConvolution3DWithSourceTensor:inputTensor
-                                                              weightsTensor:weightTransposeTensor
-                                                                 descriptor:depthWiseConv3dDescriptor_
-                                                                       name:nil];
-          } else {
-            outputTensor = [mpsGraph convolution2DWithSourceTensor:inputTensor
-                                                     weightsTensor:weightTensor
-                                                        descriptor:conv2dDescriptor_
-                                                              name:nil];
-          }
+            MPSGraphTensor* biasTensor = nil;
+            MPSGraphTensor* outputTensor;
+            if(is3DConv){
+                outputTensor = [mpsGraph convolution3DWithSourceTensor:inputTensor
+                                                         weightsTensor:weightTensor
+                                                            descriptor:conv3dDescriptor_
+                                                                  name:nil];
+                
+            } else {
+                if (bias_defined) {
+                    biasTensor =
+                    native_mps::mpsGraphUnrankedPlaceHolder(mpsGraph, native_mps::getMPSDataType(bias_opt.value()));
+                }
+                
+                if (isDepthwiseConv) {
+                    MPSGraphTensor* weightTransposeTensor = [mpsGraph transposeTensor:weightTensor
+                                                                            dimension:-3
+                                                                        withDimension:-4
+                                                                                 name:nil];
+                    outputTensor = [mpsGraph depthwiseConvolution3DWithSourceTensor:inputTensor
+                                                                      weightsTensor:weightTransposeTensor
+                                                                         descriptor:depthWiseConv3dDescriptor_
+                                                                               name:nil];
+                } else {
+                    outputTensor = [mpsGraph convolution2DWithSourceTensor:inputTensor
+                                                             weightsTensor:weightTensor
+                                                                descriptor:conv2dDescriptor_
+                                                                      name:nil];
+                }
+            }
```

Comment:
In line 319 below you want to update the `biasPlaceholder` to support 3d convolutions. I think something like this should work:

```
if (is3DConv) {
    biasPlaceholder =
        native_mps::Placeholder(cachedGraph->biasTensor_, (bias_opt.value()).view({1, bias_shape[0], 1, 1, 1}));
} else {
    biasPlaceholder =
        native_mps::Placeholder(cachedGraph->biasTensor_, (bias_opt.value()).view({1, bias_shape[0], 1, 1}));
}
```

Reply:
@Berzeg Thanks a lot for your feedback. Yes you are absolutely right, the bias shape was wrong for 3D and is now fixed